Reformat migrations and implement cleaner dry run behavior

eggplantzzz opened the pull request:
Remove `skipDryRun` from the props in the config so that it can be set on the command line as a flag.

eggplantzzz commented:
I removed the getter/setter bit of code from the `props` object. Please let me know if you think this might cause problems somewhere else in the codebase.

eggplantzzz commented:
You should still be able to set that option in the config. In the `detect` method it will still merge in `static_config` which is the values from the Truffle config.

One thing I've noted is the inconsistency with how the config is being initialized with each command. In other words, sometimes the config is initialized with `default`, some with `detect`, and others with `default().with({ ... })`. As long as the command initializes the config with something like `detect`, it should import all of the Truffle config settings. In this particular case, however, it is most likely fine since `--skip-dry-run` only applies to migrations.

CruzMolina commented:
Hmm, well after demoing the branch locally, it appears I'm unable to set the option in the config.

Can you add a test or verify on your end?

gnidan commented:
(removing myself from request for review cause y'all have it covered :)

eggplantzzz commented:
@CruzMolina So currently it will skip the dry run if you configure the variable in camel case but it will not accept it in kebab case like you would specify from the command line. It think yargs does some magic to make both versions of the property internally available to Truffle. I'll make some changes in order to handle both version.

eggplantzzz commented:
@CruzMolina I think the behavior you mentioned before is all cleared up. I also reworked a bunch of the functions in truffle-migrate to not take the callbacks that they were previously passed. I pulled it all "high up" in the command flow. If I changed any logic it was unintentional. Could you look at it with an eye out for that?

CruzMolina commented on packages/truffle-core/lib/commands/migrate.js:
```diff
@@ -110,6 +112,62 @@ var command = {
       61717561 // Aquachain
     ];
 
+    let dryRunOnly;
```
Not sure if the above actually helps resolve the issue.

eggplantzzz replied:
Yep, perfect. That was the culprit. Good catch!

CruzMolina replied:
Hmm, still the same issue locally.

eggplantzzz replied:
@CruzMolina Ok, I think I found a problem with the way a callback was being called. It looks good on my local, can you verify one more time for me please?

CruzMolina commented:
Hooray! 👏
